codemod/spotlight_util.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2007-2008 Facebook
+
+from Cocoa import *
+
+
+class SpotlightUtil(object):
+    """
+    Searches a directory hierarchy using OS/X spotlight
+    """
+
+    @staticmethod
+    def walk_directory(root_directory, match_pattern):
+        query = NSMetadataQuery.alloc().init()
+        # going to trust that the user specified the proper
+        # spotlight format pattern
+        predicate = NSPredicate.predicateWithFormat_(
+            "(kMDItemTextContent = \"" + match_pattern + "\")")
+        if root_directory is not None:
+            query.setSearchScopes_([root_directory])
+        query.setPredicate_(predicate)
+        query.startQuery()
+        # TODO: figure out how to do this asynchronously if possible
+        while query.isGathering():
```

Comment:
The most interesting part is using NSNotificationCenter to handle the end of query instead of periodically checking for it and then stopping the runloop explicitly.